### ISSUE TITLE
How initial embedding is generated?

### ISSUE THREAD

**Issue by code-rex1** [graph4ai/graph4nlp#535]
## ❓ Questions and Help

If I use the following configuration:

```
sequential_link: false
```

Then how the initial embedding would be generated? Will it still learn the embedding assuming the node order reserves the sequential order in raw text. Or `sequential_link` needs to be set to `true` for that?

**Comment by AlanSwift** [comment]
Q: Then how the initial embedding would be generated?
A: Please refer to the [graph embedding initilization](https://github.com/graph4ai/graph4nlp/blob/master/graph4nlp/pytorch/modules/graph_embedding_initialization/graph_embedding_initialization.py).
Q: Will it still learn the embedding assuming the node order reserves the sequential order in raw text. Or sequential_link needs to be set to true for that?
A: If you use LSTM to initialize the node embedding, it will reserve. Important: The sequential edges will be added into the graph when ``sequential_link`` is True.

**Comment by code-rex1** [comment]
@AlanSwift just to summarize:

1. `sequential_link` has nothing to do with the node embedding.
2. When I use `LSTM` to initialize the node embedding, it will always preserve the `sequential order in raw text`.
3. And if `sequential_link` is `True`, sequential edges will be added between sequential tokens.

Please correct me if I misunderstood.

**Comment by AlanSwift** [comment]
1. Wrong. ``sequential_link `` will add sequential order edges in the constructed graph. The node embedding after GNN calculation will contain sequential order information if it is ``True``.
2. Not exactly. The GNN takes the initial embedding A and output embedding B. A will contain sequential order if you use RNN to initialize it. And there is no guarantee that B must contain sequential order if ``sequential link`` is ``False``. Because the graph topology will not contain sequential edges if you set it false.
3. Correct.

In summary: information flow: X (token ids) -->embedding initialization: LSTM --> A --> GNN (graph topology G) --> B
1. the ``sequential_link`` is true --> G contains sequential order edges or false --> no
2. X -->A can have sequential information if you use RNN to initialize the embedding (we guarantee the inputs of RNN have the same order as raw texts).

**Comment by code-rex1** [comment]
@AlanSwift is there a graph showing the flow that you mentioned? That would be really helpful to understand the flow.

**Comment by AlanSwift** [comment]
I think this graph is coming soon. The issue will be closed.
